app/src/main/java/org/mozilla/fenix/share/ShareFragment.kt

```diff
@@ -70,6 +79,38 @@ class ShareFragment : AppCompatDialogFragment() {
         setStyle(STYLE_NO_TITLE, R.style.ShareDialogStyle)
     }
 
+    private val networkCallback = object : ConnectivityManager.NetworkCallback() {
+        override fun onLost(network: Network?) {
+            reloadDevices()
+        }
+
+        override fun onAvailable(network: Network?) {
+            reloadDevices()
+        }
+    }
+
+    private fun reloadDevices() {
+        context?.let {
+            val fxaAccountManager = it.components.backgroundServices.accountManager
+            val list = lifecycleScope.async(Dispatchers.IO) {
+                val refreshDevicesAsync =
+                    fxaAccountManager.authenticatedAccount()?.deviceConstellation()
+                        ?.refreshDevicesAsync()
+                refreshDevicesAsync?.await()
+                buildDeviceList(fxaAccountManager)
```

Comment:
`buildDeviceList` just constructs a device list for the UI to consume so it doesn't need to be done on the IO dispatcher, but I guess that's fine as well.

Please leave some comments on why we're doing this here though for when someone else comes in to read this.

Reply:
Okay took it off the IO and seems to work and be simpler :) thanks!